src/wms/src/main/java/org/geoserver/wms/legendgraphic/GetLegendGraphicKvpReader.java

```diff
@@ -728,12 +730,15 @@ private Style findStyle(String styleName, Style[] styles) throws NoSuchElementEx
     private Style[] loadRemoteStyle(String sldUrl) throws ServiceException {
         try {
             URL url = new URL(sldUrl);
+            URLCheckers.confirm(url);
```

Comment:
Check string directly before creating URL, since URLCheckers.confirm(URL) just calls out to external form ...
```suggestion
            URLCheckers.confirm(sldUrl);
            URL url = new URL(sldUrl);
```

Reply:
Note the change of order, to do the confirm(sldUrl) before creating a new URL, is on purpose.

See https://osgeo-org.atlassian.net/browse/GEOS-10384 for context.

Reply:
I made the change.